pkg/authorization/delegated/authorizer.go

```diff
@@ -30,14 +30,14 @@ import (
 	"k8s.io/klog/v2"
 )
 
-type DelegatedAuthorizerFactory func(clusterName logicalcluster.Name, client kubeclient.ClusterInterface) (authorizer.Authorizer, error)
+type DelegatedAuthorizerFactory func(clusterName logicalcluster.Name, client kubeclient.Interface) (authorizer.Authorizer, error)
 
 // NewDelegatedAuthorizer returns a new authorizer for use in e.g. admission plugins that delegates
 // to the kube API server via SubjectAccessReview.
-func NewDelegatedAuthorizer(clusterName logicalcluster.Name, client kubeclient.ClusterInterface) (authorizer.Authorizer, error) {
+func NewDelegatedAuthorizer(clusterName logicalcluster.Name, client kubeclient.Interface) (authorizer.Authorizer, error) {
```

Comment:
🔴  this signature is a blocker. What is `clusterName` now used for? We cannot store the clusterName and at the same time expect a context with the same clusterName. That is hard to reason about will definitely break somewhere someday.

Reply:
actually something is missing here. Compare line 81. There is already some cluster aware custom client. Does line 81 also need a rewrite with context?

Reply:
`clusterName` is used to scope the RESTClient calls. In Line 81, we are setting the cluster name to the [request](https://pkg.go.dev/k8s.io/client-go/rest#Request) directly. Actually, we aren't expecting the context with the clusterName at all in the DelegatedAuthorizer. If we see here, passing context without cluster name (https://github.com/kcp-dev/kcp/blob/87335b07e67c2be69e3e367a544f14f1dec6df62/pkg/admission/apibinding/apibinding_admission.go#L200) while calling `Authorize` does work.